[Content Fragment] Non-deterministic behavior of `HashMap` might fail tests using `ContentFragment.getExportedElements()`

<!-- 

Before sending a bug report please check if the latest release (if you're not already using it) fixes your problem. Alternatively, you could
install the latest snapshot version from the master branch.
 -->

## Bug Report

**Current Behavior**

- The `ContentFragmentImplTest` mocks JSON resources to `ContentFragment` objects through `ContentFragmentMockAdapter` at https://github.com/adobe/aem-core-wcm-components/blob/3524e33424bf6831faca014284c5bad803d3744d/bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/AbstractContentFragmentTest.java#L124-L125
https://github.com/adobe/aem-core-wcm-components/blob/3524e33424bf6831faca014284c5bad803d3744d/bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/AbstractContentFragmentTest.java#L170-L171
- The adapter mocks the `DAMContentElement`, which is part of `ContentFragment`, by loading the json resource at https://github.com/adobe/aem-core-wcm-components/blob/3524e33424bf6831faca014284c5bad803d3744d/bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentMockAdapter.java#L109-L116
- During the translation of `JSON` to `MockElement` using `org.apache.sling` dependency (a detailed explanation is provided with screenshots in the last section), the ordering of the properties is lost because of `HashMap.entrySet()`. According to the [Javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html), `HashMap` is not guaranteed to maintain order over time.
- This might cause issues where the order is strictly enforced.
- Specifically, the following tests expects the order of `DAMContentFragment.getExportedElementsOrder()` to be `main` followed by `second`
 1) `com.adobe.cq.wcm.core.components.internal.models.v1.contentfragment.ContentFragmentImplTest.structured`
 2) `com.adobe.cq.wcm.core.components.internal.models.v1.contentfragment.ContentFragmentImplTest.structuredNonExistingVariation`
 3) `com.adobe.cq.wcm.core.components.internal.models.v1.contentfragment.ContentFragmentImplTest.structuredVariation`
 4) `com.adobe.cq.wcm.core.components.internal.servlets.contentfragment.ElementsDataSourceServletTest.testComponentPathStructured`
 5) `com.adobe.cq.wcm.core.components.internal.servlets.contentfragment.ElementsDataSourceServletTest.testFragmentPathOverride`
 6) `com.adobe.cq.wcm.core.components.internal.servlets.contentfragment.ElementsDataSourceServletTest.testFragmentPathStructured`

#### Steps to Reproduce 
- When the above tests are run through [NonDex](https://github.com/TestingResearchIllinois/NonDex), which shuffles the order of elements returned by `HashMap.keySet()`, we could observe the mentioned issue.
```
# Clone the Repo
https://github.com/adobe/aem-core-wcm-components.git

# Compile the module byte-buddy-dep
mvn clean install -pl bundles/core -am -DskipTests

# (Optional) Run the unit test
mvn -pl bundles/core test -Dtest=<Above Mentioned Test Name>

# Run the unit test using NonDex
mvn -pl bundles/core edu.illinois:nondex-maven-plugin:2.1.1:nondex -Dtest=<Above Mentioned Test Name>
```
- Reproducing the above steps for Tests 1, 2, and 3, will produce the following result:
```
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ContentFragmentImplTest.structured:137->assertContentFragment:283->assertContentFragment:311 Element has wrong name ==> expected: <main> but was: <second>
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
```
- Reproducing the above steps for Tests 4, 5, and 6 will produce the following result:
```
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ElementsDataSourceServletTest.testComponentPathStructured:134->AbstractContentFragmentDataSourceServletTest.assertDataSource:127 expected: <main> but was: <second>
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
```

**Expected behavior/code**
- For any amount of runs, the order of `DAMContentFragment.getExportedElementsOrder()` should have been
  1) MAIN
  2) SECOND
- The test should have passed successfully without any errors with `NonDex`.

**Environment**
- Core Components version: `2.23.5-SNAPSHOT`
- Java version: `Java(TM) SE Runtime Environment (build 1.8.0_381-b09)`

**Possible Solution**
<!--- If you have suggestions for the bug fix. Please also consider creating a Pull Request 🤘. -->

- Since the change in ordering of elements loaded by `resource.getChild()` when mocking the `ContentFragment` causes the issue, sorting the elements loaded by the resource will solve the issue. Sorting the elements based on the `resource.getValueMap().keySet()` ensures that the elements be returned in the order `MAIN` followed by `SECOND`.
- Below is the diff needed to solve the above mentioned issue.
```
-/+ com.adobe.cq.wcm.core.components.internal.models.v1.contentfragment.ContentFragmentMockAdapter.java | LineNo: 111
+    for (String name : master.getValueMap().keySet().stream().sorted().collect(Collectors.toList()))
-     for (String name : master.getValueMap().keySet())
```
- This change is made only in the `ContentFragmentMockAdapter.java` because the main issue results from incorrectly loading the test data.
- If the proposed solution looks good, I can create a PR.

**Additional context / Screenshots**
<!--- Add any other context about the problem here. If applicable, add screenshots to help explain. -->
Here's an example scenario where the change in ordering occurs when run through `NonDex` for the test `com.adobe.cq.wcm.core.components.internal.models.v1.contentfragment.ContentFragmentImplTest.structured`. Since the issue occurs when loading the json resource for all the tests, the below scenario applies to all the tests.

1) The resource loaded will be specific for each test. However, one of the common files is `bundles/core/src/test/resources/contentfragment/test-content-dam-contentfragments.json`. This is the JSON from which the mocks are loaded and asserted.
<img width="561" alt="Screenshot 2023-10-31 at 7 19 14 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/9a925174-8614-4369-81e6-647359e02abb">

2) The contents of the JSON are parsed recursively at apache sling's `JsonContentParser.java:87`.  The contents are loaded from the resource into `JsonObject`. This object uses `LinkedHashMap` to store the tree structure. So, the issue is not in the JsonObject. Looking in the below image, we could observe that `object.keySet()` gives the elements in the order of the above image. However, each json object is parsed to the resource contents by extracting the values to `properties` map, which in turn is a `HashMap`. The order of values in this `HashMap` might differ based on its implementation in a specific environment or java version.

<img width="1037" alt="Screenshot 2023-10-31 at 2 33 40 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/489e6e2c-5a03-4469-8f3f-66739c112213">


<img width="497" alt="Screenshot 2023-10-31 at 2 35 05 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/75d9d664-ac6d-4eac-a1ff-027a0ef9f26d">

3) While converting the values of the json to their respective resources, the `properties` map populated in the above imaged is passed to the `LoaderContentHandler.java:82`. Since this is a `HashMap`, `content.entrySet()` might return not return the same order for all the runs. This non-deterministic order map is provider to the `JcrResourceProvider`, which converts the json values to `JcrValueMap` used by the `ContentFragmentMockAdapter`

<img width="844" alt="Screenshot 2023-10-31 at 2 37 40 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/aafa74de-8c1f-41ac-99a8-344962f905cd">

4) Accessing this map's key set at `ContentFragmentMockAdapter` gives the  following order
<img width="670" alt="Screenshot 2023-10-31 at 2 40 38 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/d9262c8f-0bf3-4b55-aba2-bde48faa9d08">

5) The mock elements created based on the above order of map is `MAIN` followed by `SECOND`
<img width="985" alt="Screenshot 2023-10-31 at 2 55 18 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/f1ba8884-2c23-4a1a-b7e7-877f5913a9c2">



#### This is the second run for the same method run through `NonDex`.


1) The order of elements from the `JsonObject` remains constant because of `LinkedHashMap`
<img width="884" alt="Screenshot 2023-10-31 at 2 43 57 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/67bd7ecf-ad50-4590-8e0b-96d118302a7d">

2) However, when the `properties` hash map is inserted, the change in order could be observed.
<img width="530" alt="Screenshot 2023-10-31 at 2 44 30 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/7fe7aef5-480b-4f8e-a0a1-d132c7c44dc7">

3) Creating the `JcrValueMap` based on this ordering
<img width="819" alt="Screenshot 2023-10-31 at 2 46 21 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/5338fded-115f-4adf-a961-5f6d3b654afb">


4) Accessing this map's key set at `ContentFragmentMockAdapter` gives a different order
<img width="658" alt="Screenshot 2023-10-31 at 2 47 14 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/0dd3d4ab-8382-4100-9801-6d2c93c793ea">

5) The mock elements created based on the above order of map is `SECOND` followed by `MAIN`
<img width="1046" alt="Screenshot 2023-10-31 at 2 54 40 PM" src="https://github.com/adobe/aem-core-wcm-components/assets/46614118/7addc19e-a693-4999-b60a-373806c2b3b9">

--- a/bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentMockAdapter.java
+++ b/bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentMockAdapter.java
@@ -106,7 +106,7 @@ public class ContentFragmentMockAdapter implements Function<Resource, ContentFra
             model = resource.getResourceResolver().getResource(data.getValueMap().get(PN_MODEL, String.class));
             // for the 'adaptTo' mock below we use the jcr:content child to mimick the real behavior
             modelAdaptee = model.getChild(JCR_CONTENT);
-            // create an element mock for each property on the master node
+            // create an element mock for each property on the master node sorted by property name
             Resource master = resource.getChild(PATH_MASTER);
             for (String name : master.getValueMap().keySet().stream().sorted().collect(Collectors.toList())) {
                 // skip the primary type and content type properties

--- a/bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentMockAdapter.java
+++ b/bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/contentfragment/ContentFragmentMockAdapter.java
@@ -23,6 +23,7 @@ import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
 import java.util.function.Function;
+import java.util.stream.Collectors;
 
 import javax.annotation.Nullable;
 
@@ -107,7 +108,7 @@ public class ContentFragmentMockAdapter implements Function<Resource, ContentFra
             modelAdaptee = model.getChild(JCR_CONTENT);
             // create an element mock for each property on the master node
             Resource master = resource.getChild(PATH_MASTER);
-            for (String name : master.getValueMap().keySet()) {
+            for (String name : master.getValueMap().keySet().stream().sorted().collect(Collectors.toList())) {
                 // skip the primary type and content type properties
                 if (JcrConstants.JCR_PRIMARYTYPE.equals(name) || name.endsWith("@ContentType")) {
                     continue;

